utils/ingress/wrapper.go

```diff
@@ -149,6 +173,85 @@ func (i *Ingress) SetAnnotations(annotations map[string]string) {
 	}
 }
 
+func (i *Ingress) CreateAnnotationBasedPath(actionName string) {
+	i.mux.Lock()
+	defer i.mux.Unlock()
+	if HasRuleWithService(i, actionName) {
+		return
+	}
+	switch i.mode {
+	case IngressModeNetworking:
+		t := v1.PathTypeImplementationSpecific
+		p := v1.HTTPIngressPath{
+			Path:     "/*",
+			PathType: &t,
+			Backend: v1.IngressBackend{
+				Service: &v1.IngressServiceBackend{
+					Name: actionName,
+					Port: v1.ServiceBackendPort{
+						Name: "use-annotation",
+					},
+				},
+			},
+		}
+		for _, rule := range i.ingress.Spec.Rules {
+			rule.HTTP.Paths = append(rule.HTTP.Paths[:1], rule.HTTP.Paths[0:]...)
+			rule.HTTP.Paths[0] = p
```

Comment:
This should be in the order of managedRoutes pretty sure if this is the route for headers we should place it on top of any existing path but ordered by managedRoutes order.

Reply:
Thanks! I will fix it.